docs/topics/development/waffle.md

```diff
@@ -0,0 +1,127 @@
+# Waffle
+
+We use [waffle](https://waffle.readthedocs.io/en/stable/) for managing feature access in production.
+
+## Why switches and not flags
+
+We prefer to use [switches](https://waffle.readthedocs.io/en/stable/types/switch.html)
+over flags, as switches are:
+
+- switches are simple
+- switches are easy to reason about
+
+## Creating/Deleting a switch
+
+Switches are added via database migrations.
+This ensures the switch exists in all environments once the migration is run.
+
+To create or remove a switch,
+first create an empty migration in the app where your switch will live.
+
+```bash
+python ./manage.py makemigrations <app> --empty
+```
+
+### Creating a switch
+
+add the switch in the migration
+
+```python
+# Generated by Django 4.2.8 on 2024-01-09 11:18
+
+from django.db import migrations
+
+from olympia.core.db.migrations import CreateWaffleSwitch
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('devhub', '0005_auto_20220531_1043'),
+    ]
+
+    operations = [
+        CreateWaffleSwitch('foo')
+    ]
+```
+
+### Deleting a switch
+
+remove the switch in the migration
+
+```python
+# Generated by Django 4.2.8 on 2024-01-09 11:18
+
+from django.db import migrations
+
+from olympia.core.db.migrations import DeleteWaffleSwitch
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('devhub', '0005_auto_20220531_1043'),
+    ]
+
+    operations = [
+        DeleteWaffleSwitch('foo')
+    ]
+```
+
+## Using a switch
+
+Use your switch in python code
+
+```python
+if waffle.switch_is_active('foo'):
+    # do something
+```
+
+Use your switch in jinja2
+
+```django
+{% if waffle.switch_is_active('foo') %}
+    <p>foo is active</p>
+{% endif %}
+```
+
+## Testing
+
+Testing the result of a switch being on or off is important
+to ensure your switch behaves appropriately. We can override the value of a switch easily.
+
+Override for an entire test case
+
+```python
+# Override an entire test case class
+@override_switch('foo', active=True)
+class TestFoo(TestCase):
+    def test_bar(self):
+        assert waffle.switch_is_active('foo')
+
+    # Override an individual test method
+    @override_switch('foo', active=False)
+    def test_baz(self):
+        assert not waffle.switch_is_active('foo')
+```
+
+## Enabling your switch in production
+
+Once your switch is deployed, you can enable in a given environment by following these steps.
+
+1. ssh into a kubernetes pod in the environment you want to enable the switch in. ([instructions][devops])
+2. run the CLI command to enable your switch ([instructions][waffle-cli])
+
+Toggling a switch on
+
+```bash
+./manage.py waffle_switch foo on
+```
+
```

Comment:
Once you've ensured that it works on dev, the typical way of doing things would be to add that `manage.py` command to the deploy instructions for the relevant tag. The engineer responsible for the tag would run the command on stage, then SRE would run it in production on deploy.

Reply:
Done.